wagtail_transfer/views.py

```diff
@@ -190,6 +239,57 @@ def do_import(request):
         return redirect('wagtailadmin_explore_root')
 
 
+def import_model(request):
+    source = request.POST['source']
+    model = request.POST['source_model']
+    base_url = settings.WAGTAILTRANSFER_SOURCES[source]['BASE_URL']
+    digest = digest_for_source(source, model)
+
+    url = f"{base_url}api/models/{model}/"
+    if request.POST.get("source_model_object_id"):
+        source_model_object_id = request.POST.get("source_model_object_id")
+        url = f"{url}{source_model_object_id}/"
+
+    response = requests.get(url, params={'digest': digest})
+    importer = ImportPlanner(model=model)
+    importer.add_json(response.content)
+
+    while importer.missing_object_data:
```

Comment:
Seems to be mostly the same as the previous view, could we extract the shared bits into separate functions?

Reply:
Oh yes, good catch. I had meant to do this originally.